cmd/storageos/storageos.go

```diff
@@ -35,6 +37,38 @@ func init() {
 	debug.Disable()
 }
 
+func isCoreOS() (bool, error) {
+	f, err := ioutil.ReadFile("/etc/lsb-release")
+	if err != nil {
+		return false, err
+	}
+
+	return strings.Contains(string(f), "DISTRIB_ID=CoreOS"), nil
+}
+
+func isInContainer() (bool, error) {
+	f, err := ioutil.ReadFile("/proc/1/cgroup")
+	if err != nil {
+		return false, err
+	}
+
+	// TODO: How reliable is this method of detection. Is there a better way?
+	return strings.Contains(string(f), "docker"), nil
+}
+
+func verfyHostPlatform() error {
+	// Detect native execution on coreOS
+	// coreOS should not run user-land programs, and we will not work there (outside a container)
+	if coreos, err := isCoreOS(); err == nil && coreos {
+
+		// If we dont think we are in a container, fail and warn the user
+		if inContainer, err := isInContainer(); err == nil && !inContainer {
+			return errors.New("do not run user-land utilities directly on coreOS, please use the CLI container")
```

Comment:
To use the StorageOS CLI on Container Linux, run:

docker run --rm -e STORAGEOS_HOST -e STORAGEOS_USERNAME -e STORAGEOS_PASSWORD storageos/cli

Reply:
@oicheryl After a chat with Simon, it seems that we are not really set-up for supporting multi-line errors. I agree that your message is far better, however it is outside the scope of this PR to add proper support for multi-line messages.

I am happy to take a ticket for this feature should one exist (be created).

Reply:
Then I would replace with "To use the StorageOS CLI on Container Linux, you need to run the storageos/cli image."

Reply:
I have made this change, I will treat this PR as merge-able.